### PR TITLE
Add search scores, models, and deployments to "Thought process" tab, surface additional properties

### PR DESCRIPTION
**Pull request by pamelafox**
## Purpose

This PR was inspired by a conversation with @charris-msft when we were reviewing evaluation results. It adds more information to the thought process tab: 

* model & deployment for any LLM steps
* search scores (including reranker_score) inside search results
* additional search properties that weren't shown before
* the full prompt used in the query generation stage (this does make the tab visibly longer, but I think it's helpful for developers to really grok what's happening)

Screenshots:

Chat:
Screenshot 2024-03-06 at 7 12 46 AM
Screenshot 2024-03-06 at 7 12 33 AM

Chat with vision:
Screenshot 2024-03-06 at 7 13 47 AM

Ask:
Screenshot 2024-03-06 at 7 11 03 AM

Ask with vision:
Screenshot 2024-03-06 at 7 11 59 AM

## Does this introduce a breaking change?

When developers merge from main and run the server, azd up, or azd deploy, will this produce an error?
If you're not sure, try it out on an old environment.

```
[ ] Yes
[X] No
```

## Does this require changes to learn.microsoft.com docs?

This repository is referenced by [this tutorial](https://learn.microsoft.com/azure/developer/python/get-started-app-chat-template)
which includes deployment, settings and usage instructions. If text or screenshot need to change in the tutorial,
check the box below and notify the tutorial author. A Microsoft employee can do this for you if you're an external contributor.

```
[ ] Yes
[X] No
```

## Type of change

```
[ ] Bugfix
[X] Feature
[ ] Code style update (formatting, local variables)
[ ] Refactoring (no functional changes, no api changes)
[ ] Documentation content changes
[ ] Other... Please describe:
```

## Code quality checklist

See [CONTRIBUTING.md](../CONTRIBUTING.md#submit-pr) for more details.

- [X] The current tests all pass (`python -m pytest`).
- [X] I added tests that prove my fix is effective or that my feature works - snapshot tests cover this.
- [X] I ran `python -m pytest --cov` to verify 100% coverage of added lines
- [X] I ran `python -m mypy` to check for type errors
- [X] I either used the pre-commit hooks or ran `ruff` and `black` manually on my code.

### CODE REVIEW

**Review by charris-msft — APPROVED** [review]
Love it!

**Comment by pamelafox** [comment]
@charris-msft Thanks for reviewing! I actually needed this in my live stream today where I was talking about evaluation. I used the debugger instead and the score thresholds were super low for the irrelevant content. So am excited to see it easily surfaced in the UI. Will await review from @mattgotteiner since this is also a code change.

File: app/frontend/src/components/AnalysisPanel/AnalysisPanel.tsx [hunk 1]
```diff
@@ -1,5 +1,4 @@
 import { Stack, Pivot, PivotItem } from "@fluentui/react";
-import SyntaxHighlighter from "react-syntax-highlighter";
```
Comment (mattgotteiner) [inline app/frontend/src/components/AnalysisPanel/AnalysisPanel.tsx:2]: Surprised we ever needed this

**Review by mattgotteiner — APPROVED** [review]
Thanks!

File: app/frontend/src/components/AnalysisPanel/AnalysisPanel.tsx (hunk 1 above)
Comment (pamelafox) [reply app/frontend/src/components/AnalysisPanel/AnalysisPanel.tsx:2]: That's used in the ThoughtProcess component, which moved into its own file. It makes our JSON slightly prettier than normal, in theory.

**Pull request merged**